src/service/viewer-impl.js

```diff
@@ -951,6 +930,31 @@ export class Viewer {
   whenMessagingReady() {
     return this.messagingMaybePromise_;
   }
+
+  /**
+   * Replace the
+   * @param {string} newUrl
+   */
+  replaceUrl(newUrl) {
+    if (!this.ampdoc.isSingleDoc() || !this.win.history.replaceState) {
```

Comment:
should we keep the `|| !newUrl` so it does not behave differently than before for the case where result is empty string?

Reply:
I check that `newUrl` exist before calling the function. It should make no difference right? But anyway I agree it is better to check inside the function, and saves a type convert. 
added the check to the function.